### PR TITLE
Fixes #31785 - update pulp rpm

### PR DESCRIPTION
**Pull request by jjeffers**
This PR updates pulp_rpm to version 3.9.0.

### CODE REVIEW

**Comment by jjeffers** [comment]
[test katello]

**Comment by jjeffers** [comment]
[test katello]

**Comment by jjeffers** [comment]
[test katello]

**Comment by jjeffers** [comment]
[test katello]

File: katello.gemspec [hunk 1]
```diff
@@ -57,8 +57,8 @@ Gem::Specification.new do |gem|
   gem.add_dependency "pulp_ansible_client", ">= 0.2", "< 0.7"
   gem.add_dependency "pulp_container_client", ">= 2.0.0", "< 2.3.0"
   gem.add_dependency "pulp_deb_client", ">= 2.6.0", "< 2.9.0"
-  gem.add_dependency "pulp_rpm_client", ">=3.6.2", "< 3.9.0"
-  gem.add_dependency "pulp_2to3_migration_client", ">= 0.7.0", "< 0.8.0"
+  gem.add_dependency "pulp_rpm_client", ">=3.9.0", "< 3.9.1"
+  gem.add_dependency "pulp_2to3_migration_client", ">= 0.3.0", "< 0.7.0", "!= 0.4.0"
```
Comment (jlsherrill) [inline katello.gemspec:61]: I think this was a bad conflict resolution (the 2to3 migration line change)

**Comment by jjeffers** [comment]
[test katello]

**Comment by jjeffers** [comment]
[test katello]

**Comment by jjeffers** [comment]
[test katello]

File: katello.gemspec [hunk 2]
```diff
@@ -57,8 +57,8 @@ Gem::Specification.new do |gem|
   gem.add_dependency "pulp_ansible_client", ">= 0.2", "< 0.7"
   gem.add_dependency "pulp_container_client", ">= 2.0.0", "< 2.3.0"
   gem.add_dependency "pulp_deb_client", ">= 2.6.0", "< 2.9.0"
-  gem.add_dependency "pulp_rpm_client", ">=3.6.2", "< 3.9.0"
-  gem.add_dependency "pulp_2to3_migration_client", ">= 0.7.0", "< 0.8.0"
+  gem.add_dependency "pulp_rpm_client", ">=3.9.0", "< 3.9.1"
```
Comment (ianballou) [inline katello.gemspec:60]: I'm curious about why the pulp_rpm_client version is locked to 3.9.0.  Or did you mean this? 
```ruby
  gem.add_dependency "pulp_rpm_client", ">=3.9.0", "< 3.10.0"
```

**Comment by ianballou** [comment]
Other than my gemspec comment it looks fine to me. Are we waiting for pulp-rpm 3.9 to be packaged in the Pulpcore 3.9 repo before merging?

File: katello.gemspec (hunk 2 above)
Comment (jjeffers) [reply katello.gemspec:60]: Actually that is not intentional, and I suppose that should be changed.

**Comment by jlsherrill** [comment]
i opened some packaging prs: 
 https://github.com/theforeman/pulpcore-packaging/pull/94
https://github.com/theforeman/pulpcore-packaging/pull/93

we can merge when they are ready to go

**Comment by jjeffers** [comment]
[test katello]

**Comment by jjeffers** [comment]
[test katello]

**Comment by jjeffers** [comment]
[test katello]

File: katello.gemspec [hunk 3]
```diff
@@ -57,8 +57,8 @@ Gem::Specification.new do |gem|
   gem.add_dependency "pulp_ansible_client", ">= 0.2", "< 0.7"
   gem.add_dependency "pulp_container_client", ">= 2.0.0", "< 2.3.0"
   gem.add_dependency "pulp_deb_client", ">= 2.6.0", "< 2.9.0"
-  gem.add_dependency "pulp_rpm_client", ">=3.6.2", "< 3.9.0"
-  gem.add_dependency "pulp_2to3_migration_client", ">= 0.7.0", "< 0.8.0"
+  gem.add_dependency "pulp_rpm_client", ">=3.9.0", "< 3.10.0"
+  gem.add_dependency "pulp_2to3_migration_client", ">= 0.3.0", ">= 0.7.0", "< 0.8.0"
```
Comment (ianballou) [inline katello.gemspec:61]: Looks like `pulp_2to3_migration_client` got changed accidentally.

**Review by ianballou — APPROVED** [review]
Looks good!

**Pull request merged**